PageNumbersWatermark is broken

The PageNumbersWatermark test  (com.lowagie.examples.directcontent.pageevents.PageNumbersWatermark) breaks with this exception:

```
com.lowagie.examples.directcontent.pageevents.PageNumbersWatermark
com.lowagie.text.DocumentException: com.lowagie.text.exceptions.IllegalPdfSyntaxException: Unbalanced save/restore state operators.
	at com.lowagie.text.pdf.PdfDocument.add(PdfDocument.java:775)
	at com.lowagie.text.Document.add(Document.java:276)
	at com.lowagie.examples.directcontent.pageevents.PageNumbersWatermark.main(PageNumbersWatermark.java:71)
Caused by: com.lowagie.text.exceptions.IllegalPdfSyntaxException: Unbalanced save/restore state operators.
	at com.lowagie.text.pdf.PdfContentByte.sanityCheck(PdfContentByte.java:3148)
	at com.lowagie.examples.directcontent.pageevents.PageNumbersWatermark.onEndPage(PageNumbersWatermark.java:166)
	at com.lowagie.text.pdf.PdfDocument.newPage(PdfDocument.java:866)
	at com.lowagie.text.pdf.PdfDocument.carriageReturn(PdfDocument.java:1245)
	at com.lowagie.text.pdf.PdfDocument.add(PdfDocument.java:547)
	... 2 more

```

--- a/pdf-toolbox/src/test/java/com/lowagie/examples/directcontent/pageevents/PageNumbersWatermark.java
+++ b/pdf-toolbox/src/test/java/com/lowagie/examples/directcontent/pageevents/PageNumbersWatermark.java
@@ -161,8 +161,8 @@ public class PageNumbersWatermark extends PdfPageEventHelper {
             catch(Exception e) {
                 throw new ExceptionConverter(e);
             }
-            cb.restoreState();
         }
+        cb.restoreState();
         cb.sanityCheck();
     }
     

